### PR TITLE
Unable to create detached pool

### LINKED ISSUE #155: Unable to create detached pool
#### Agent Version

master (this is a bug for Mitaka release)
#### Operating System

CentOS 7
#### OpenStack Release

Mitaka
#### Description

New feature in Mitaka release allows the operator to create a pool and not bind it to a listener.  The F5 agent throws an error when attempting this operation.
#### Deployment

Single OpenStack controller with one F5 agent managing 1 BIG-IP VE

### PR DESCRIPTION
**Pull request by jlongstaf**
@richbrowne, @pjbreaux 
#### What issues does this address?
Fixes #155 

#### What's this change do?
Refactors service builder to create a list of pools independent of listeners. Previously, only pools attached to a listener were included in the service object.

#### Where should the reviewer start?
service_builder.py

#### Any background context?
No

Issues:
Fixes #155

Problem: Unable to create a pool not bound to a listener.

Analysis: Refactored service builder to create list of pools
independent of listeners.

Tests: unit test, tempest functional test

### CODE REVIEW

File: f5lbaasdriver/test/tempest/tests/api/test_service_builder.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+        # get an RPC client for calling into driver
+        cls.client = cls.plugin_rpc.get_client()
+        cls.context = cls.plugin_rpc.get_context()
+
+    @classmethod
+    def resource_cleanup(cls):
+        super(ServiceBuilderTestJSON, cls).resource_cleanup()
+
+    def check_service_object(self):
```
Comment (pjbreaux) [inline f5lbaasdriver/test/tempest/tests/api/test_service_builder.py]: The standard in OpenStack is to name the tests with prefix 'test_'

Comment (pjbreaux) [inline f5lbaasdriver/test/tempest/tests/api/test_service_builder.py]: Nevermind, got it

File: f5lbaasdriver/v2/bigip/service_builder.py [hunk 2]
```diff
@@ -456,3 +408,92 @@ def _get_l7policy_rules(self, context, l7policies):
...
+
+    @log_helpers.log_method_call
+    def _get_listeners(self, context, loadbalancer):
+        listeners = []
+        db_listeners = self.plugin.db.get_listeners(
+            context,
+            filters={'loadbalancer_id': [loadbalancer.id]}
+        )
+
+        for listener in db_listeners:
```
Comment (jlongstaf) [inline f5lbaasdriver/v2/bigip/service_builder.py]: Provisioning and operating status now included.

File: f5lbaasdriver/v2/bigip/service_builder.py [hunk 3]
```diff
@@ -456,3 +408,92 @@ def _get_l7policy_rules(self, context, l7policies):
...
+                )
+
+                subnet_map[subnet['id']] = subnet
+                network_map[network['id']] = network
+                pool_members.append(member_dict)
+
+        return pool_members
+
+    @log_helpers.log_method_call
+    def _pool_to_dict(self, pool):
```
Comment (jlongstaf) [inline f5lbaasdriver/v2/bigip/service_builder.py]: Provisioning and operating status now included.

**Comment by jlongstaf** [comment]
Final result for service object:

[service_obj.txt](https://github.com/F5Networks/f5-openstack-lbaasv2-driver/files/556908/service_obj.txt)

**Comment by pjbreaux** [comment]
That service object looks really clean.

**Pull request merged**